### DISCUSSION TITLE
Using rtsp with another source

### DISCUSSION THREAD

**Discussion by augustodinizl** [bluenviron/mediamtx#379] [Questions] [answered]
Greetings, 

I have a doubt, and was wondering if here https://github.com/aler9/rtsp-simple-server#on-demand-publishing i can use another program as video source, i was planning to use this one https://github.com/arthar360/TaniDVR as source for the above code, is this currently doable?

**Comment by aler9** [comment, marked as answer]
Hello, you should be able to stream from a DVR to rtsp-simple-server by combining together TaniDVR and ffmpeg in this way

```
tanidvr -m 1 -c 1 -m 1 -t host -u user -w pass | ffmpeg -r 25 -i - -c copy -f rtsp rtsp://localhost:8554/stream
```

Unfortunately i don't own a DVR and i can't test it.

**Reply by augustodinizl** [reply]
> Hello, you should be able to stream from a DVR to rtsp-simple-server by combining together TaniDVR and ffmpeg in this way
> 
> ```
> tanidvr -m 1 -c 1 -m 1 -t host -u user -w pass | ffmpeg -r 25 -i - -c copy -f rtsp rtsp://localhost:8554/stream
> ```
> 
> Unfortunately i don't own a DVR and i can't test it.

Dude, thanks for your fast response, that helped a lot, it's almost there, i've got a error now (read it here: https://github.com/augustodinizl/NotePad/blob/master/log%20tanidvr) i'll dig about that...

**Reply by augustodinizl** [reply]
well, after some time, i think i'm "sorting it out" with some "tweaks"... i think the path is that, i've changed the rtsp-simple-server.yml with that code:
`paths:
  ondemand:
      runOnDemand: tanidvr -m 1 -c 1 -t 192.168.0.108 -u admin -w ADMIN -f c2.mkv | ffmpeg -re -stream_loop -1 -i c2.mkv -c copy -f rtsp rtsp://localhost:$RTSP_PORT/$RTSP_PATH
    runOnDemandRestart: yes`

and i get some stream, but after some time, it hangs...
just for information, here the output from the server:
https://github.com/augustodinizl/NotePad/blob/master/rtspsemi-successlog.txt

Any advices will be appreciated

**Reply by augustodinizl** [reply]
@aler9 , SUCCESS!!! i've got it to work with live stream without using the file output (.mkv)
now i will try to close the pipe when it get the ondemand closes signal...
for documentation, i've used the following line inside rtsp-simple-server.yml:
'streamX:
    runOnDemand: tanidvr -m 1 -t 192.168.0.xxx -u user -w password -c X | ffmpeg -r 25 -i - -c copy -f rtsp rtsp://localhost:$RTSP_PORT/$RTSP_PATH
    runOnDemandRestart: yes'

**Reply by augustodinizl** [reply]
@aler9 it is possible to send a kill process signal to rtsp-simple-server when i get the ondemand feed closed?

**Reply by aler9** [reply]
Add killall at the end of the command:

```yml
streamX:
  runOnDemand: tanidvr -m 1 -t 192.168.0.xxx -u user -w password -c X | ffmpeg -r 25 -i - -c copy -f rtsp rtsp://localhost:$RTSP_PORT/$RTSP_PATH; killall rtsp-simple-server
```

**Reply by augustodinizl** [reply]
unfortunatelly, there's no effect on that, tanidvr is still running when the feed is closed, sorry, i did'nt make myself clear about that issue, rtsp-simple server works fine, tanidvr and ffmpeg aren't closing

**Reply by augustodinizl** [reply]
Well, i couldn't been able to achieve the full features that i wanted, but hey, it's open-source! there's always a way, since i'm using the rtsp-simple-server to stream a feed to a home assistant instance, i've been able to use ssh to send the needed commands to the host, so, so far so good... again, MANY THANKS @aler9 for the awesome work, and for the availabillity!

**Comment by DBiao** [comment]
![image](https://user-images.githubusercontent.com/19355201/123760691-add4f780-d8f3-11eb-8acb-77cf40897152.png)
How do you parse the header data from []byte

[a comment by an automated account is not shown]
